custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/BuildSettings.java

```diff
@@ -37,6 +42,9 @@
     private DockerBuildSettings docker;
     @CheckForNull
     private JenkinsfileRunnerSettings jenkinsfileRunner;
+    @CheckForNull
+    private String updateCenterUrl;
```

Comment:
P2: I'm not seeing any integration tests where a value for the new `updateCenterUrl` parameter/setting is provided.  Seeing as there are getters and setters for all the other fields in this class, I'm wondering if there's magic involved (the value will be automatically deserialized without needing to implement a getter/setter pair) or simply an omission.

Reply:
There is no stable update center I could use for such test. Expoerimental Update Center (https://jenkins.io/doc/developer/publishing/releasing-experimental-updates/) could be used, but it would be matter of time till test assumptions change (e.g. an experimental-only plugin gets released). Setting up a test update center is doable, but it is an overkill IMHO. I can create a follow-up JIRA ticket

Reply:
I added a smoke test with a TODO comment for now

Reply:
It might be sufficient to use wiremock to set up a simple localhost endpoint that returns some unique JSON payload, just to confirm that the de-serialization is taking place correctly and the custom value can be used.